miden-lib/asm/sat/kernel.masm

```diff
@@ -5,13 +5,41 @@ use.miden::sat::internal::layout
 use.miden::sat::internal::note
 use.miden::sat::internal::tx
 
+# AUTHENTICATION
+# =================================================================================================
+
+#! Authenticates that the invocation of a kernel procedure originates from the user context.
+#!
+#! Panics:
+#!   - if the invocation of the kernel procedure does not originate from the user context.
+#!
+#! Stack: [...]
+#! Output: [...]
+proc.authenticate_user_origin
```

Comment:
minor nit: I think the term "user" here is a bit confusing as notes are also executed in user context. Maybe a better name would be `authenticate_account_origin` or maybe `authenticate_account_procedure` to indicate that only procedures belonging to an account interface can pass this check?